persistence/remote/src/test/java/org/infinispan/persistence/remote/RemoteStoreTest.java

```diff
@@ -38,45 +44,101 @@
+   public Object[] factory() {
+      return new Object[] {
+            new RemoteStoreTest().segmented(false).cacheMediaType(MediaType.APPLICATION_OBJECT).rawValues(true),
+            new RemoteStoreTest().segmented(false).cacheMediaType(MediaType.APPLICATION_OBJECT).rawValues(false),
+            new RemoteStoreTest().segmented(false).cacheMediaType(MediaType.APPLICATION_PROTOSTREAM).rawValues(true),
+            new RemoteStoreTest().segmented(false).cacheMediaType(MediaType.APPLICATION_PROTOSTREAM).rawValues(false),
+            new RemoteStoreTest().segmented(true).cacheMediaType(MediaType.APPLICATION_OBJECT).rawValues(true),
+            new RemoteStoreTest().segmented(true).cacheMediaType(MediaType.APPLICATION_OBJECT).rawValues(false),
+            new RemoteStoreTest().segmented(true).cacheMediaType(MediaType.APPLICATION_PROTOSTREAM).rawValues(true),
+            new RemoteStoreTest().segmented(true).cacheMediaType(MediaType.APPLICATION_PROTOSTREAM).rawValues(false),
+      };
+   }
 
    @Override
-   protected AdvancedLoadWriteStore createStore() {
-      ConfigurationBuilder localBuilder = TestCacheManagerFactory.getDefaultCacheConfiguration(false);
-      localBuilder.memory().evictionType(EvictionType.COUNT).size(WRITE_DELETE_BATCH_MAX_ENTRIES).expiration().wakeUpInterval(10L);
+   protected String parameters() {
+      return "[" + segmented + ", " + cacheMediaType + ", " + isRawValues + "]";
+   }
+
+   @Override
+   protected Configuration buildConfig(ConfigurationBuilder cb) {
+      cb.memory().maxCount(WRITE_DELETE_BATCH_MAX_ENTRIES)
+            .expiration().wakeUpInterval(10L);
+
+      // Unfortunately BaseNonBlockingStore stops and restarts the store, which can start a second hrServer - prevent that
+      if (hrServer == null) {
+         GlobalConfigurationBuilder globalConfig = new GlobalConfigurationBuilder().clusteredDefault();
+         globalConfig.defaultCacheName(REMOTE_CACHE);
+
+         ConfigurationBuilder configurationBuilder = hotRodCacheConfiguration(cb);
+         configurationBuilder.encoding().mediaType(cacheMediaType.toString());
+         configurationBuilder.clustering().cacheMode(CacheMode.DIST_SYNC);
+         localCacheManager = TestCacheManagerFactory.createClusteredCacheManager(
+               globalConfig, configurationBuilder);
+         TestingUtil.replaceComponent(localCacheManager, TimeService.class, timeService, true);
+
+         localCache = (AdvancedCache<Object, Object>) localCacheManager.getCache(REMOTE_CACHE).getAdvancedCache()
+               .withMediaType(MediaType.APPLICATION_OBJECT_TYPE, MediaType.APPLICATION_OBJECT_TYPE);
+         keyPartitioner = localCache.getAdvancedCache().getComponentRegistry().getComponent(KeyPartitioner.class);
+
+         hrServer = HotRodClientTestingUtil.startHotRodServer(localCacheManager);
+         // In case if the server has to unmarshall the value, make sure to use the same marshaller
+//         hrServer.setMarshaller(getMarshaller());
```

Comment:
Remove?

Reply:
Sure, I was more testing to see if it was needed, which it obviously isn't :)